routers/web/admin/users.go

```diff
@@ -209,7 +209,11 @@ func NewUserPost(ctx *context.Context) {
 func prepareUserInfo(ctx *context.Context) *user_model.User {
 	u, err := user_model.GetUserByID(ctx.ParamsInt64(":userid"))
 	if err != nil {
-		ctx.ServerError("GetUserByID", err)
+		if user_model.IsErrUserNotExist(err) {
+			ctx.Redirect(setting.AppSubURL + "/admin/users")
```

Comment:
That seems wrong.
Why would you redirect here to the admin page?
I thought you perhaps wanted to redirect to their new name just as GitHub does, but that is not an improvement.
Especially because normal users should not be redirected to admin URLs.

Reply:
Another possible quickfix would be to display 404 instead.
That one I could live with.
But I'm strongly against this implementation.

Reply:
We are in the admin section and no user names are involved only the id.

Reply:
Oh, true.
But still, is redirecting without a preceding `404` really a good idea?

Reply:
Lots of apps just redirect you in such a case because it does not disrupt the "ui flow". But for me a not found error is fine too.